### PR TITLE
Fork build trigger config and deployment script for custom DC autopush

### PR DESCRIPTION
**Pull request by hqpho**
- cloudbuild.deploy.yaml -> cloudbuild.push.yaml, which adds a step to update repo version and removes a step to build and push a custom DC image
- deploy shell script -> build-and-deploy shell script, which pulls latest submodules and builds/pushes a new image before deploying.

Once the new trigger flow is fully set up, the old yaml and script can be deleted.

### CODE REVIEW

File: build/ci/cloudbuild.push.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    args:
+      - -c
+      - |
+        set -e
+        gcloud source repos clone deployment /tmp/deployment --project=datcom-ci
+        cd /tmp/deployment
+        ./scripts/update_external_repo_version.sh $REPO_NAME $SHORT_SHA
+    waitFor: ["-"]
+
+  - id: push-web-server
```
Comment (shifucun) [inline build/ci/cloudbuild.push.yaml:28]: also waitFor: ["-"] for "push-web-server"?

File: build/ci/cloudbuild.push.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        set -e
+        docker build -f build/website_cron_testing/Dockerfile \
+          -t gcr.io/datcom-ci/website-cron-testing:$SHORT_SHA \
+          -t gcr.io/datcom-ci/website-cron-testing:latest \
+          .
+        docker push gcr.io/datcom-ci/website-cron-testing:$SHORT_SHA
+        docker push gcr.io/datcom-ci/website-cron-testing:latest\
+    waitFor: ["-"]
+
+  - id: update version
```
Comment (shifucun) [inline build/ci/cloudbuild.push.yaml:87]: maybe add some comments for this one (and differentiate with update-repo-version)

**Review by shifucun — APPROVED** [review]
very nice!!

File: build/ci/cloudbuild.push.yaml (hunk 2 above)
Comment (hqpho) [reply build/ci/cloudbuild.push.yaml:87]: Renamed and added comments.

File: build/ci/cloudbuild.push.yaml (hunk 1 above)
Comment (hqpho) [reply build/ci/cloudbuild.push.yaml:28]: Done.

**Pull request merged**